deep_folding/brainvisa/utils/resample.py

```diff
@@ -112,36 +123,137 @@ def resample(input_image: Union[str, aims.Volume],
     # if values is not None, values are given in ascending order
     # Note also that background shall not be given in values
 
-    # Create one bucket by value (except background)
-    # FIXME: Create several buckets because I didn't understood how to add
-    # several bucket to a BucketMap
-    for i, v in enumerate(values):
+    if aims.version() >= (5, 2):
+        reorder = False
         toc = time()
+        if values != sorted(np.unique(vol_dt[vol_dt != background])):
```

Comment:
To clarify the comparison between the two arrays, I think that np.any() is needed.
In the case where the two arrays have different lengths, the comparison will raise an error in recent numpy versions (for previous versions of numpy, it is only a warning)